.github/workflows/build.yml

```diff
@@ -52,15 +63,26 @@ jobs:
             maven-build-cache-
       - name: Install Poetry
         uses: snok/install-poetry@v1
+      - name: Poetry cache
+        id: cached-poetry
+        uses: actions/cache@v4
+        with:
+          path: ~/.cache/pypoetry
+          key: poetry-cache-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            poetry-
+      - name: Install Helm
+        run: |
+          curl -fsSL -o get_helm.sh https://raw.githubusercontent.com/helm/helm/master/scripts/get-helm-3
+          chmod 700 get_helm.sh
+          ./get_helm.sh
```

Comment:
Q: Just curious if there was any success on the cache size limitation

Reply:
I wasn't able to identify a good approach to keep shrink our cache. However, I did run across the [documentation](https://docs.github.com/en/enterprise-server@3.9/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#configuring-cache-storage-for-a-repository) that enables repo owner to increase cache size. At the moment, 25GB is sufficient for our build.